cmd/zc_enumerator.go

```diff
@@ -243,10 +259,18 @@ func initResourceTraverser(source string, location common.Location, ctx *context
 			return nil, err
 		}
 
-		if ctx == nil || p == nil {
+		// check if credential is also nil here (would never trigger) to tame syntax highlighting.
+		// As a precondition to pipeline p, credential must not be nil anyway.
+		if ctx == nil || p == nil || credential == nil {
 			return nil, errors.New("a valid credential and context must be supplied to create a blobFS traverser")
 		}
 
+		if sourceURL.Scheme == "http" && credential.CredentialType != common.ECredentialType.SharedKey() {
```

Comment:
Is in necessary to have this exception for shared key auth?  IIRC Shared Key auth does protect the credentials over HTTP (which is presumably why this exception is here)… but it does not protect the data that's in transit (request/response body) nor does it protect filenames (request headers).  So I'm feeling that we should always warn, even if using SharedKey.

Reply:
I tackled this due to a PBI regarding if we should support HTTP outside of SharedKey usage... I'm not entirely sure why, but I think @zezha-msft has said that SharedKey is actually better on HTTP for some reason.

Reply:
It is better than the other forms of auth, for the reason I said in my comment.  But, also for the reasons I said in my comment, it's not better _enough_ for us to justify providing no warning.

Reply:
Fair point. I'll change this.